[25.05] Patch FRR to stub out neighbour table management

sysvinit opened the pull request:
We've been having recurring issues on the routers where after a failover the secondary router will continue to announce stuck MAC-IP routes, which will cause other hardware machines to import incorrect MAC/IP mappings into their neighbour tables. This commit introduces a further patch to FRR which disables the neighbour table management functions used in EVPN to avoid the symptoms of this problem as we haven't been able to fix the source.

PL-132588

@flyingcircusio/release-managers

## Release process

- [x] Created changelog entry using `./changelog.sh` => none, internal change

## PR release workflow (internal)

- [x] PR has internal ticket
- [x] internal issue ID (PL-…) part of branch name
- [x] internal issue ID mentioned in PR description text
- [x] ticket is on Platform agile board
- [ ] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team
<!---->
- [x] set urgency and risk labels
- [x] ensure the merge bot has determined a merge date
- [x] ensure all checks are green
- [x] get a review from a colleague

## Design notes

- [ ] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [ ] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - This patch changes the intended behaviour of FRR to diverge from what is expected and tested upstream. However, this only disables functionality, which reduces the amount of code which is actually executed.
- [x] Security requirements tested? (EVIDENCE)
  - Tested without noticeable regressions in both dev and on a handful of production hosts which have been hitting the neighbour table corruption issues particularly frequently.

sysvinit committed:
pkgs: patch frr to stub out neighbour table management

FRR's neighbour table management is a frequent source of headaches
whose root cause we're not well placed to fix properly at the
moment. Stub out the neighbour table management to work around the
symptoms instead.

PL-132588

osnyx reviewed (changes requested):
tests.frr.migration fails with

```
subtest: check remote TAP responder is reachable
host2: must succeed: ping -A -c5 192.168.23.101
host2: output: PING 192.168.23.101 (192.168.23.101) 56(84) bytes of data.

--- 192.168.23.101 ping statistics ---
5 packets transmitted, 0 received, 100% packet loss, time 4005ms


!!! Test "check remote TAP responder is reachable" failed with error: "command `ping -A -c5 192.168.23.101` failed (exit code 1)"
!!! Test "migrate guest TAP responder" failed with error: "command `ping -A -c5 192.168.23.101` failed (exit code 1)"
!!! Traceback (most recent call last):
!!!   File "<string>", line 55, in <module>
!!!     host2.succeed(f"ping -A -c5 {guest_ip}")
!!! 
!!! RequestedAssertionFailed: command `ping -A -c5 192.168.23.101` failed (exit code 1)
cleanup
```

sysvinit committed:
tests: trim down comments to reflect reality

PL-132588

sysvinit committed:
pkgs: fix gratuitous arp in ping-on-tap

RFC 5227 Section 1.1 requires that in ARP Annoucements, the sender
hardware address in the ARP packet payload (which is distinct from the
source address in the ethernet frame header!) be set to the address of
the interface sending the packet.

PL-132588
